homeassistant/components/switchbee/cover.py

```diff
@@ -23,59 +23,91 @@
 from .coordinator import SwitchBeeCoordinator
 from .entity import SwitchBeeDeviceEntity
 
-_DeviceTypeT = TypeVar(
-    "_DeviceTypeT",
-    bound=Union[SwitchBeeShutter, SwitchBeeSomfy],
-)
-
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up SwitchBee switch."""
     coordinator: SwitchBeeCoordinator = hass.data[DOMAIN][entry.entry_id]
     async_add_entities(
-        SwitchBeeCover(device, coordinator)
+        SwitchBeeCoverEntity(device, coordinator)
         for device in coordinator.data.values()
         if isinstance(
             device,
-            (
-                SwitchBeeShutter,
-                SwitchBeeSomfy,
-            ),
+            SwitchBeeShutter,
         )
     )
 
+    async_add_entities(
+        SwitchBeeSomfyEntity(device, coordinator)
+        for device in coordinator.data.values()
+        if isinstance(
+            device,
+            SwitchBeeSomfy,
+        )
+    )
+
+
+class SwitchBeeSomfyEntity(SwitchBeeDeviceEntity[SwitchBeeSomfy], CoverEntity):
+    """Representation of a SwitchBee Somfy cover."""
+
+    _attr_device_class = CoverDeviceClass.SHUTTER
+
+    def __init__(
+        self,
+        device: SwitchBeeSomfy,
+        coordinator: SwitchBeeCoordinator,
+    ) -> None:
+        """Initialize the SwitchBee cover."""
+        super().__init__(device, coordinator)
+        self._attr_current_cover_position = 0
+        self._attr_is_closed = True
```

Comment:
I'm not sure if these two are any use anymore. Maybe look at `attr_assumed_state`.

Reply:
Unable to find examples

Reply:
Since both are unknown, I think you should:
- set `_attr_is_closed = None` as a class attribute, because `CoverEntity` doesn't have a default.
- remove `_attr_current_cover_position` and leave the default value `None` from `CoverEntity`
- and at this stage you should be able to remove the `init` method since it doesn't do any customisation anymore